include/posix/pthread.h

```diff
@@ -24,6 +23,16 @@ static inline s32_t _ts_to_ms(const struct timespec *to)
 	return (to->tv_sec * 1000) + (to->tv_nsec / 1000000);
 }
 
+typedef struct pthread_sem {
+        struct k_sem *sem;
```

Comment:
since there are no other fields besides the semaphore kernel object, have you considered just typedefing sem_t to struct k_sem?

Reply:
I followed Andy's method to implement posix APIs. He did same for pthread_mutex_t. Do you suggest to do this other way ?